src/components/Sliding/index.js

```diff
@@ -0,0 +1,226 @@
+    <path class="sliding__controls__image__path" d="M1.16675 6H12.8334M12.8334 6L7.83341 11M12.8334 6L7.83341 1" stroke="#1B85F3" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
+    </svg>
+  </button>
+</div>`;
+
+export default function Sliding({
+  items = [],
+  loopMode = 'normal',
+  controls = true,
+} = {}) {
+  Component.call(this, { html, events });
+
+  this.loopMode = loopMode;
+  this.controls = controls;
+  this.items = [];
+
+  items.forEach((item) => this.addItem(item));
+
+  this.updateItems();
+
+  this.setControls(controls);
+
+  this.selected.get('previous-button').disabled =
+    this.getLoopMode() === 'normal';
+
+  makeSwipable(this.selected.get('sliding-content'));
+
+  initializeSwiper();
+
+  this.selected
+    .get('sliding-content')
+    .addEventListener('swipe-left', () => this.next());
+  this.selected
+    .get('sliding-content')
+    .addEventListener('swipe-right', () => this.prev());
+
+  this.selected
+    .get('previous-button')
+    .addEventListener('click', () => this.prev());
+
+  this.selected.get('next-button').addEventListener('click', () => this.next());
+
+  this.listen('mount', () =>
+    document.addEventListener('click', this.setSlide(this.items[0])),
+  );
+
+  this.listen('unmount', () =>
+    document.addEventListener('click', this.clearItems()),
+  );
+
+  this.getCurrentIndex = () =>
+    this.items.findIndex((item) =>
+      item.classList.contains('sliding__content__slide--selected'),
+    );
+
+  this.addNextAnimation = () => {
+    const slide = this.getSlide();
+
+    slide.classList.remove('sliding__content__slide__animation--prev');
+    slide.classList.add('sliding__content__slide__animation--next');
+  };
+
+  this.addPrevAnimation = () => {
+    const slide = this.getSlide();
+
+    slide.classList.remove('sliding__content__slide__animation--next');
+    slide.classList.add('sliding__content__slide__animation--prev');
+  };
+
+  this.updateStatusControls = () => {
+    const currentIndex = this.getCurrentIndex();
+
+    this.selected.get('previous-button').disabled = currentIndex === 0;
+
+    this.selected.get('next-button').disabled =
+      currentIndex === this.items.length - 1;
+
+    this.emit('controls:disabled');
+  };
+}
+
+Sliding.prototype = Object.assign(Sliding.prototype, Component.prototype, {
+  addItem(item) {
+    item.classList.add('sliding__content__slide');
+    this.selected.get('sliding-content').appendChild(item);
+
+    this.updateItems();
+
+    this.emit('item:add', item);
+  },
+
+  removeItem(item) {
+    const itemExist = this.selected.get('sliding-content').contains(item);
+
+    if (!itemExist) throw new Error('Item not exists');
+
+    this.selected.get('sliding-content').removeChild(item);
+
+    this.updateItems();
+
+    this.emit('item:remove', item);
+  },
+
+  setSlide(item) {
+    this.items.forEach((i) => {

```

Comment:
essas variaveis de uma letra é meio paia, nao fica muito claro o que é
```suggestion
    this.items.forEach((item) => {

```